patchwork-fml/src/main/java/net/minecraftforge/fml/InterModComms.java

```
@@ -0,0 +1,188 @@
+	/**
+	 * Retrieve pending messages for your modid. Use the predicate to filter the method name.
+	 *
+	 * @param modId the modid you are querying for
+	 * @param methodMatcher a predicate for the method you are interested in
+	 * @return All messages passing the supplied method predicate
+	 */
+	public static Stream<IMCMessage> getMessages(final String modId, final Predicate<String> methodMatcher) {
+		ConcurrentLinkedQueue<IMCMessage> queue = containerQueues.get(modId);
+
+		if (queue == null) {
+			return Stream.empty();
+		}
+
+		return StreamSupport.stream(new QueueFilteringSpliterator(queue, methodMatcher), false);
+	}
+
+	/**
+	 * Retrieve all message for your modid.
+	 *
+	 * @param modId the modid you are querying for
+	 * @return All messages
+	 */
+	public static Stream<IMCMessage> getMessages(final String modId) {
+		return getMessages(modId, s -> Boolean.TRUE);
+	}
+
+	private static class QueueFilteringSpliterator implements Spliterator<IMCMessage> {
+		private final ConcurrentLinkedQueue<IMCMessage> queue;
+		private final Predicate<String> methodFilter;
+		private final Iterator<IMCMessage> iterator;
+
+		private QueueFilteringSpliterator(final ConcurrentLinkedQueue<IMCMessage> queue, final Predicate<String> methodFilter) {
+			this.queue = queue;
+			this.iterator = queue.iterator();
+			this.methodFilter = methodFilter;
+		}
+
+		@Override
+		public int characteristics() {
+			return Spliterator.CONCURRENT | Spliterator.NONNULL | Spliterator.ORDERED;
+		}
+
+		@Override
+		public long estimateSize() {
+			return queue.size();
+		}
+
+		@Override
+		public boolean tryAdvance(final Consumer<? super IMCMessage> action) {
+			IMCMessage next;
+
+			do {
+				if (!iterator.hasNext()) {
+					return false;
+				}
+
+				next = this.iterator.next();
+			}
+
+			while (!methodFilter.test(next.method));
+
+			action.accept(next);
+			this.iterator.remove();
+			return true;
+		}
+
+		@Override
+		public Spliterator<IMCMessage> trySplit() {
+			return null;
```

throw an UnsupportedOperationException instead of silently returning null